src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -50,6 +55,17 @@ public static void Initialize()
         // as you can't make a constructor partial.
         private RuntimeEventSource(int _) { }
 
+        private enum EventId : int
+        {
+            AppContextSwitch = 1
+        }
+
+        [Event((int)EventId.AppContextSwitch, Level = EventLevel.Informational, Keywords = Keywords.AppContext)]
+        internal void LogAppContextSwitch(string switchName, int value)
```

Comment:
@brianrob Would a single event with all of the data be better for something like this?

Reply:
I think one event per switch is good.  It makes it easy to consume the data, and I can't imagine we'll have so many that the cost of this event will be painful.  If/when we choose to create events were we dump data in bulk, it makes both the event production and event consumption code more complex.  On the production side, you have to chunk the data to ensure that you don't run up against size limitations (e.g. ETW has a 64K size limit per event).  On the consumption side, you have to be more sophisticated in how you parse the event.  There are definitely places were it's valuable - for example BulkType events in the runtime when we log heap snapshots.  There are just tons of types, and so emitting each one in its own event is very costly.